lisa/sut_orchestrator/azure/arm_template.bicep

```diff
@@ -0,0 +1,323 @@
+}
+
+func getCreateDisk(disk object, diskName string, index int) object => {
+  name: diskName
+  createOption: disk.create_option
+  caching: disk.caching_type
+  diskSizeGB: disk.size
+  lun: index
+  managedDisk: {
+      storageAccountType: disk.type
+  }
+}
+
+func getOsDiskVhd(vmName string) object => {
+  id: resourceId('Microsoft.Compute/images', '${vmName}-image')
+}
+
+func getLinuxConfiguration(keyPath string, publicKeyData string, disable_password_authentication bool) object => {
+  disablePasswordAuthentication: disable_password_authentication
+  ssh: {
+      publicKeys: [
+          {
+              path: keyPath
+              keyData: publicKeyData
+          }
+      ]
+  }
+  provisionVMAgent: true
+}
+
+func getEphemeralOSImage(node object) object => {
+  name: '${node.name}-osDisk'
+  diffDiskSettings: {
+      option: 'local'
+      placement: 'CacheDisk'
+  }
+  caching: 'ReadOnly'
+  createOption: 'FromImage'
+  diskSizeGB: node.osdisk_size_in_gb
+}
+
+func getAttachDisk(disk object, diskName string, index int) object => {
+  lun: index
+  createOption: 'attach'
+  caching: disk.caching_type
+  managedDisk: {
+      id: resourceId('Microsoft.Compute/disks', diskName)
+  }
+}
+
+func getOsDiskSharedGallery(node object) object => {
+  id: resourceId(node.subscription_id, empty(node.resource_group_name) ? 'None' : node.resource_group_name, 'Microsoft.Compute/galleries/images/versions', node.image_gallery, node.image_definition, node.image_version)
+}
+
+func getOsDiskMarketplace(node object) object => {
+  publisher: node.marketplace.publisher
+  offer: node.marketplace.offer
+  sku: node.marketplace.sku
+  version: node.marketplace.version
+}
+
+func getSPWithDiskEncryptionSet(node object) object => {
+  securityEncryptionType: node.security_profile.encryption_type
+  diskEncryptionSet: {
+    id: node.security_profile.disk_encryption_set_id
+  }
+}
+
+func getSP(node object) object => {
+  securityEncryptionType: node.security_profile.encryption_type
+}
+
+func getOSImage(node object) object => {
+  name: '${node.name}-osDisk'
+  managedDisk: {
+    storageAccountType: node.os_disk_type
+    securityProfile: ((empty(node.security_profile) || (node.security_profile.security_type != 'ConfidentialVM')) ? null : (empty(node.security_profile.disk_encryption_set_id) ? getSP(node) : getSPWithDiskEncryptionSet(node)))
+  } 
+  caching: ((node.os_disk_type == 'Ephemeral') ? 'ReadOnly' : 'ReadWrite')
+  createOption: 'FromImage'
+  diskSizeGB: node.osdisk_size_in_gb
+}
+
+func getSecurityProfile(security_profile object) object => {
+  uefiSettings: {
+    secureBootEnabled: security_profile.secure_boot
+    vTpmEnabled: true
+  }
+  securityType: security_profile.security_type
+}
+
+func getAvailabilitySetId(availability_set_name string) object => {
+  id: resourceId('Microsoft.Compute/availabilitySets', availability_set_name)
+}
+
+func hasVmgsPath(node object) bool => bool((!empty(node.vhd)) && (!empty(node.vhd.vmgs_path)))
```

Comment:
It should be called `isCvm`. The function names can be used to explain the business logic.

Reply:
updated